src/v0/destinations/salesforce/transform.js

```diff
@@ -123,7 +124,19 @@ async function getSaleforceIdForRecord(
       destination.ID,
     );
   }
-  return get(processedsfSearchResponse.response, 'searchRecords.0.Id');
+  const searchRecord = processedsfSearchResponse.response?.searchRecords?.find(
+    (rec) => rec[identifierType] === identifierValue,
+  );
+  if (!searchRecord) {
+    logger.error(
+      `[SALESFORCE] URL for fetching searchRecord: /services/data/v${SF_API_VERSION}/parameterizedSearch/?q=${identifierValue}&sobject=${objectType}&in=${identifierType}&${objectType}.fields=id,${identifierType}`,
```

Comment:
could this have any PII data like emails etc?

Reply:
No, the only parameters part of this be custom object names (in salesforce) or external_id field name. Shouldn't be PII.